masschallenge/impact-api
# [AC-4857] - OPTIONS from descriptions

Done:
- Revised OPTIONS call for /v1/user, /v1/user/<id>, /v1/organization, /v1/ogranization/<id>, /v1/industry, /v1/industry/<id>, /v1/program_family, /v1/program_family/<id>, /v1/user/<id>/history, /v1/organization/<id>/history, /v1/user/<id>/organizations, /v1/organization/<id>/users, /v1/user/<id>/confidential.
- All these calls are now based on a description of the call that is then used to derive the OPTIONS response.  This description is expected to be leveraged for other calls (GET, POST, PATCH), but that has been deferred to a follow on ticket.
- This has been made reasonable abstract and should be pretty straight forward to support for future.  A good example is the UserConfidentialView where the needed code is captured in USER_CONFIDENTIAL_FIELDS and the metadata method.  Ideally, the CONFIDENTIAL_KEYS would be derived from USER_CONFIDENTIAL_FIELDS, but that's the next level that's been moved to a follow on ticket.  In this case it is simple, but it gets hairy for stuff like /v1/user as well as POST and PATCH.

To test:
- Load up the code with a fresh clean db and compare and contrast the results/arguments to the calls with what OPTIONS claims.  In Postman (collection is https://www.getpostman.com/collections/8df2bd6e977c11a3b776) you can get the OPTIONS for each call by temporarily changing the dropdown next to the URL to 'OPTIONS'.  We should probably add these calls to the Postman collection, but didn't have the time to do that for everything. **Y.M: An updated Postman collection with OPTIONS [here](https://www.getpostman.com/collections/0a95773f88df764b15ee)**
